### PR TITLE
Lint the `modules.json` for uninstalled modules

### PR DESCRIPTION
**Pull request by KevinMenden**
This PR adds a new pipeline lint check (`nf-core lint`) which looks for module entries in the `modules.json` of modules that are not actually installed in the pipeline.

I thought it makes sense to package that into the `nf-core lint` command as it looks at a general pipeline file (`modules.json`) and compares that against all installed modules. The `nf-core modules lint` command is more module-centric, meaning that it runs specific lint tests for every module separately. Therefore I think it is more logical to add this check here.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/lint/modules_json.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    modules_json = module_remove.load_modules_json()
+
+    if modules_json:
+        module_remove.get_pipeline_modules()
+
+        all_modules_passed = True
+
+        for key in modules_json["modules"].keys():
+            if not key in module_remove.module_names:
+                failed.append(f"Entry for`{key}` found in `modules.json` but module is not installed in pipeline.")
```
Comment (ErikDanielsson) [inline nf_core/lint/modules_json.py:30]:
```suggestion
                failed.append(f"Entry for `{key}` found in `modules.json` but module is not installed in pipeline.")
```

File: nf_core/lint/modules_json.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                all_modules_passed = False
+
+        if all_modules_passed:
+            passed.append("Only installed modules found in `modules.json`")
+    else:
+        warned.append("Could not open modules.json file.")
```
Comment (ErikDanielsson) [inline nf_core/lint/modules_json.py:36]:
```suggestion
        warned.append("Could not open `modules.json` file.")
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -30,6 +30,7 @@
 * Updated `nf-core modules remove` to remove module entry in `modules.json` if module directory is missing
 * Create extra tempdir as work directory for `nf-core modules create-test-yml` to avoid adding the temporary files to the `test.yml`
 * Refactored passing of command line arguments to `nf-core` commands and subcommands ([#1139](https://github.com/nf-core/tools/issues/1139), [#1140](https://github.com/nf-core/tools/issues/1140))
+* Check for `modules.json` for entires of modules that are not actually installed in the pipeline [[#1141]](https://github.com/nf-core/tools/issues/1141)
```
Comment (ErikDanielsson) [inline CHANGELOG.md:33]:
```suggestion
* Check for `modules.json` for entries of modules that are not actually installed in the pipeline [[#1141](https://github.com/nf-core/tools/issues/1141)]
```

**Review by ErikDanielsson — APPROVED** [review]
Looks good! Just found a few minor things.

File: nf_core/lint/modules_json.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+
+from logging import warn
+from nf_core.modules.remove import ModuleRemove
```
[the remark this answers is not in the document]
Comment (KevinMenden) [reply nf_core/lint/modules_json.py:4]: yep you're right, didn't think of that :)

**Pull request merged**